### PR TITLE
[Glossary] Adding the definiton of vCPU to the Glossary

### LINKED ISSUE #1909: [Glossary] Add the definition of vCPU to the glossary

### PR DESCRIPTION
**Pull request by EditKoselak**
Closes #1909

### CODE REVIEW

File: doc/common/glossary.md [hunk 1]
```diff
@@ -20,144 +20,193 @@ To help guide the reader, this glossary provides an introduction to the terminol
...
-  - Within **NFV**, A **Network Function** is implemented in a form of **Virtualised NF** or a **Cloud Native NF**.
-- **Network Service (NS)**: composition of **Network Function**(s) and/or **Network Service**(s), defined by its functional and behavioural specification, including the service lifecycle.
-- **Virtual Network Function (VNF)**: a software implementation of a **Network Function**, capable of running on the **Cloud Infrastructure**.
-  - **VNF**s are built from one or more VNF Components (**VNFC**) and, in most cases,  the VNFC is hosted on a single VM or Container.
-- **Cloud Native Network Function (CNF)**: A cloud native network function (CNF) is a cloud native application that implements network functionality. A CNF consists of one or more microservices and has been developed using Cloud Native Principles including immutable infrastructure, declarative APIs, and a “repeatable deployment process”.
+
+- **Cloud Infrastructure Profile:** The combination of the Cloud Infrastructure Software Profile and the Cloud Infrastructure Hardware Profile that defines the capabilities of the Cloud Infrastructure.
```
Comment (pgoyal01) [inline doc/common/glossary.md:26]: I think we may need to be specify the nature of capabilities and for what; the profile also specifies certain configurations. While Cloud Infrastructure Software Profile is defined the Hardware Profile is not.
```suggestion
- **Cloud Infrastructure Profile:** The combination of the Cloud Infrastructure Software Profile and the Cloud Infrastructure Hardware Profile that defines the capabilities and configuration of the Cloud Infrastructure resources made available to workloads.
```

File: doc/common/glossary.md [hunk 2]
```diff
@@ -20,144 +20,193 @@ To help guide the reader, this glossary provides an introduction to the terminol
...
+
+- **Cloud Infrastructure Software Configuration:** a set of settings (Key:Value) that are applied/mapped to **cloud infrastructure** SW deployment.
+
+- **Cloud Infrastructure Software Profile:** defines the behaviour, capabilities and metrics provided by a Cloud Infrastructure Software Layer.
```
Comment (pgoyal01) [inline doc/common/glossary.md:30]:
```suggestion
- **Cloud Infrastructure Software Profile:** defines the behaviour, capabilities and metrics provided by a Cloud Infrastructure Software Layer on resources made available to workloads.
```

File: doc/common/glossary.md [hunk 3]
```diff
@@ -20,144 +20,193 @@ To help guide the reader, this glossary provides an introduction to the terminol
...
+  - **Control Plane Workloads:** that perform tasks related to any other communication between NFs that is not directly related to the end-to-end data communication between applications. For example, this category includes session management, routing or authentication.
+  - **Storage Workloads:** that perform tasks related to disk storage (either SSD or HDD or other).  Examples range from non-intensive router logging to more intensive database read/write operations.
 
 <a name="1.2"></a>
 ## Hardware Layer Terminology
 
-- **Physical Network Function (PNF)**: Implementation of a network function via tightly coupled dedicated hardware and software system.
+- **Cloud Infrastructure Hardware Configuration:** a set of settings (Key:Value) that are applied/mapped to **Cloud Infrastructure** HW deployment.
+
+- **Cloud Infrastructure Hardware Profile:** defines the behaviour, capabilities, configuration, and metrics provided by a cloud infrastructure hardware layer.
```
Comment (pgoyal01) [inline doc/common/glossary.md:84]:
```suggestion
- **Cloud Infrastructure Hardware Profile:** defines the behaviour, capabilities, configuration, and metrics provided by a cloud infrastructure hardware layer resources made available to workloads.
```

File: doc/common/glossary.md [hunk 4]
```diff
@@ -20,144 +20,193 @@ To help guide the reader, this glossary provides an introduction to the terminol
...
+ >_*Note:*_ The NFVI can span across many locations, e.g. places where data centres or edge nodes are operated. The network providing connectivity between these locations is regarded to be part of the cloud infrastructure. **NFVI** and **VNF** are the top-level conceptual entities in the scope of Network Function Virtualisation. All other components are sub-entities of these two main entities.
+
+- **Network Service (NS):** composition of **Network Function**(s) and/or **Network Service**(s), defined by its functional and behavioural specification, including the service lifecycle.
+
+- **Software Defined Storage (SDS):** An architecture which consists of the storage software that is independent from the underlying storage hardware. The storage access software provides data request interfaces (APIs) and the SDS controller software provides storage access services and networking.
+
+- **Virtual Application (VA):** A general term for software which can be loaded into a Virtual Machine.
+  >_*Note:*_ a **VNF** is one type of VA.
+
+- **Virtual CPU (vCPU):** Represents a portion of the host's computing resources allocated to a workload. One or more vCPUs can be assigned to a workload.
```
Comment (pgoyal01) [inline doc/common/glossary.md:59]: technically the vCPUs are assigned to a VM or container (or a virtualised resource).
```suggestion
- **Virtual CPU (vCPU):** Represents a portion of the host's computing resources allocated to a virtualised resource (virtual machine or container). One or more vCPUs can be assigned to a  virtualised resource.
```

File: doc/common/glossary.md (hunk 3 above)
Comment (walterkozlowski) [reply doc/common/glossary.md:84]: @EditKoselak and @pgoyal01: i agree with the last version.  @EditKoselak: can you please accept the update or respond?

File: doc/common/glossary.md (hunk 4 above)
Comment (walterkozlowski) [reply doc/common/glossary.md:59]: @EditKoselak and @pgoyal01: i agree with the last version.  @EditKoselak: can you please accept the update or respond?

**Comment by walterkozlowski** [comment]
@EditKoselak and @pgoyal01: can you guys agree on the wording (see my comments) so we can go ahead with this PR?  Thanks

**Review by pgoyal01 — APPROVED** [review]

File: doc/common/glossary.md [hunk 5]
```diff
@@ -23,13 +23,13 @@ To help guide the reader, this glossary provides an introduction to the terminol
...
 >_*Note:*_ **NFVI**, **IaaS** and **CaaS** layers can be built on top of each other. In case of CaaS some cloud infrastructure features (e.g.: HW management or multitenancy) are implemented by using an underlaying **IaaS** layer.
 
-- **Cloud Infrastructure Profile:** The combination of the Cloud Infrastructure Software Profile and the Cloud Infrastructure Hardware Profile that defines the capabilities of the Cloud Infrastructure.
+- **Cloud Infrastructure Profile:** The combination of the Cloud Infrastructure Software Profile and the Cloud Infrastructure Hardware Profile that defines the capabilities and configuration of the Cloud Infrastructure resources available for the workloads.
 
 - **Cloud Infrastructure Software Configuration:** a set of settings (Key:Value) that are applied/mapped to **cloud infrastructure** SW deployment.
 
-- **Cloud Infrastructure Software Profile:** defines the behaviour, capabilities and metrics provided by a Cloud Infrastructure Software Layer.
+- **Cloud Infrastructure Software Profile:** defines the behaviour, capabilities and metrics provided by a Cloud Infrastructure Software Layer on resources available for the workloads.
```
Comment (ulikleber) [inline doc/common/glossary.md:30]: Why do we need to add this? The infrastructure defines that for any resources, also if they are unavailable.

File: doc/common/glossary.md [hunk 6]
```diff
@@ -23,13 +23,13 @@ To help guide the reader, this glossary provides an introduction to the terminol
...
 
-- **Cloud Native Network Function (CNF):** A cloud native network function (CNF) is a cloud native application that implements network functionality. A CNF consists of one or more microservices. All layers of a CNF are developed using Cloud Native Principles including immutable infrastructure, declarative APIs, and a “repeatable deployment process”.
+- **Cloud Native Network Function (CNF):** A cloud native network function (CNF) is a cloud native application that implements network functionality. A CNF consists of one or more microservices. All layers of a CNF is developed using Cloud Native Principles including immutable infrastructure, declarative APIs, and a “repeatable deployment process”.
```
Comment (ulikleber) [inline doc/common/glossary.md:32]: All layers ... is ??? 
It should stay "are".

File: doc/common/glossary.md [hunk 7]
```diff
@@ -56,6 +56,8 @@ To help guide the reader, this glossary provides an introduction to the terminol
 - **Virtual Application (VA):** A general term for software which can be loaded into a Virtual Machine.
   >_*Note:*_ a **VNF** is one type of VA.
 
+- **Virtual CPU (vCPU):** Represents a portion of the host's computing resources allocated to a virtualised resource, for example, to a virtual machine or a container. One or more vCPUs can be assigned to a  virtualised resource.
```
Comment (ulikleber) [inline doc/common/glossary.md:59]: This definition is too wide. It applies to many other things, including a physical CPU in an SMP system or a Linux process.

File: doc/common/glossary.md [hunk 8]
```diff
@@ -79,7 +81,7 @@ To help guide the reader, this glossary provides an introduction to the terminol
 
 - **Cloud Infrastructure Hardware Configuration:** a set of settings (Key:Value) that are applied/mapped to **Cloud Infrastructure** HW deployment.
 
-- **Cloud Infrastructure Hardware Profile:** defines the behaviour, capabilities, configuration, and metrics provided by a cloud infrastructure hardware layer.
+- **Cloud Infrastructure Hardware Profile:** defines the behaviour, capabilities, configuration, and metrics provided by a cloud infrastructure hardware layer resources available for the workloads.
```
Comment (ulikleber) [inline doc/common/glossary.md:84]: Why do we need to add this? The hardware profile defines that for any resources, also if they are unavailable.

**Review by ulikleber — CHANGES_REQUESTED** [review]

File: doc/common/glossary.md (hunk 5 above)
Comment (pgoyal01) [reply doc/common/glossary.md:30]: @ulikleber Is the concern about "available" then that can be dropped -- is "... on resources for workloads" better?

File: doc/common/glossary.md (hunk 8 above)
Comment (pgoyal01) [reply doc/common/glossary.md:84]: @ulikleber  is "... on resources for workloads" better?

File: doc/common/glossary.md (hunk 7 above)
Comment (CsatariGergely) [reply doc/common/glossary.md:59]: Any suggestions how to make it more narrow?

**Review by walterkozlowski — APPROVED** [review]

**Comment by walterkozlowski** [comment]
@ulikleber, @petorre : please review, we want to close this out

**Pull request merged**